src/fs/dvfs/dvfs_util.c

```diff
@@ -123,11 +124,15 @@ int dvfs_destroy_inode(struct inode *inode) {
 
 	if (inode->i_sb && inode->i_sb->sb_ops &&
 	    inode->i_sb->sb_ops->destroy_inode)
-		return inode->i_sb->sb_ops->destroy_inode(inode);
-	else
-		return dvfs_default_destroy_inode(inode);
+		inode->i_sb->sb_ops->destroy_inode(inode);
```

Comment:
Specific destroy_inode should free fs-specific resources.
It shouldn't handle inode pool, as it is hidden to them.